app/controllers/dradis/plugins/csv/upload_controller.rb

```diff
@@ -2,10 +2,23 @@ module Dradis::Plugins::CSV
   class UploadController < ::AuthenticatedController
     include ProjectScoped
 
+    before_action :load_attachment, only: [:new]
+
     def new
+      @default_columns = ['Unique Identifier', 'Column Header From File', 'Type', 'Field in Dradis']
+
+      @headers = ::CSV.open(@attachment.fullpath, &:readline)
+    end
+
+    def create
+      redirect_to main_app.project_upload_manager_path(current_project)
+    end
+
+    private
+
+    def load_attachment
```

Comment:
if I was paranoid, I'd add validation here before reading the file. Unless `Attachment` already does.

Reply:
The `Attachment` model is checking if the attachment does exist in the project. In another PR, we're also checking the validity of the CSV file and redirect accordingly